### DISCUSSION TITLE
How to draw i3s model's solid edge?

### DISCUSSION THREAD

**Discussion by zairl23** [visgl/deck.gl#7270] [Q&A]
arcgis implement : https://developers.arcgis.com/javascript/latest/sample-code/layers-scenelayer-edges/
But at present, we found that deck does not render edge, similar to arcgis no-edge effect,  so how to add solid edge effect?

**Comment by radoslavirha** [comment]
Hi @zairl23,

I tried to implement `ink` `PostProcessEffect` which does not render edges, but model looks little bit better. Some edged only solution would be great!

https://github.com/visgl/loaders.gl-showcases/issues/86

**Reply by felixpalmer** [reply]
This is not a straightforward effect to do well. None of the postprocess filters in deck do proper edge detection based on the geometry. You'll have to create your own. 

A common approach is to use the depth buffer of the scene to create the edges. I'm not sure how to access the depth buffer in an effect, but in principle it should be possible. Then you'll need a shader to read the depth values and compute the edges, you should be able to find some online e.g. https://github.com/MizunagiKB/edge_detect_shader
